livekit/sip.go

```diff
@@ -103,9 +103,24 @@ func (p *SIPOutboundTrunkInfo) AsTrunkInfo() *SIPTrunkInfo {
 }
 
 func validateHeader(header string) error {
-	v := strings.ToLower(header)
-	if !strings.HasPrefix(v, "x-") {
-		return fmt.Errorf("only X-* headers are allowed: %s", header)
+	if header == "" {
+		return errors.New("header name is empty")
+	}
+	if strings.ContainsFunc(header, func(r rune) bool {
+		if r >= 'a' && r <= 'z' {
+			return false
+		} else if r >= 'A' && r <= 'Z' {
+			return false
+		} else if r >= '0' && r <= '9' {
+			return false
+		}
+		switch r {
+		case '-':
+			return false
+		}
+		return true
+	}) {
+		return fmt.Errorf("invalid header name: %q", header)
```

Comment:
Can we replace it by sth. like:
```
if !regexp.MustCompile(`^[a-zA-Z0-9\-]*$`).MatchString(word) {
 return fmt.Errorf("invalid header name: %q", header)
}
```

Reply:
btw. I suppose the number full of `-` is also incorrect

Reply:
Done, switched to regexp. Full header validation is out of scope - SIP itself will drop invalid headers during parsing. This is just a best-effort validation to prevent typos.